backend/apps/core/utils/caching.py

```
@@ -0,0 +1,369 @@
+from rest_framework import status
+
+
+# Default cache timeout (in seconds)
+# GET requests: 5 minutes (300 seconds)
+# List/retrieve operations can be cached longer
+DEFAULT_CACHE_TIMEOUT = 300  # 5 minutes
+LIST_CACHE_TIMEOUT = 300     # 5 minutes
+DETAIL_CACHE_TIMEOUT = 600   # 10 minutes
+
+
+def generate_cache_key(request, view_name: str, include_user: bool = True, viewset_instance=None) -> str:
```

Ruff (0.14.3)

Unused function argument: `viewset_instance`

(ARG001)